src/metabase/models/persisted_info.clj

```diff
@@ -117,6 +117,14 @@
   ([conditions-map state]
    (t2/update! PersistedInfo conditions-map {:active false, :state state, :state_change_at :%now})))
 
+(defn invalidate!
+  "Invalidates any caches corresponding to the `conditions-map`. Equivalent to toggling the caching off and on again."
+  [conditions-map]
+  ;; TODO confirm that the corresponding tables will still be cleaned up.
```

Comment:
If it doesn't we should rework the caching mechanism itself, since this is no different to what you end up with if you turn it off and on again in the interface.

Reply:
It won't get cleaned up based on this:

https://github.com/metabase/metabase/blob/3c71e3a473a440e17c5fb0e90e35c5ab2e55aeee/src/metabase/task/persist_refresh.clj#L128-L130

The only states that get cleaned up are in `persisted-info/prunable-states`, i.e. `"deletable"` or `"off"` when using EE. If persisted info is in the `"creating"` state it will be persisted here

https://github.com/metabase/metabase/blob/3c71e3a473a440e17c5fb0e90e35c5ab2e55aeee/src/metabase/task/persist_refresh.clj#L60

Reply:
I see we actually use a consistent table name for the cache (`:table-name` on the `persisted-info` row), so this is actually fine - we'll clobber the stale cache during the next refresh cycle.